pandas/tests/indexes/period/test_arithmetic.py

```diff
@@ -335,102 +272,228 @@ def test_sub_isub(self):
+                                   freq='H')
+
+        result = rng - other
+        tm.assert_index_equal(result, expected)
+
+        rng -= other
+        tm.assert_index_equal(rng, expected)
+
+    def test_add_iadd_timedeltalike_annual(self):
+        # offset
+        # DateOffset
+        rng = pd.period_range('2014', '2024', freq='A')
+        result = rng + pd.offsets.YearEnd(5)
+        expected = pd.period_range('2019', '2029', freq='A')
+        tm.assert_index_equal(result, expected)
+        rng += pd.offsets.YearEnd(5)
+        tm.assert_index_equal(rng, expected)
+
+    @pytest.mark.parametrize('other', [
+        pd.offsets.YearBegin(2),
+        pd.offsets.MonthBegin(1),
+        pd.offsets.Minute(),
+        np.timedelta64(365, 'D'),
+        timedelta(365),
+        Timedelta(days=365)])
+    def test_pi_add_iadd_timedeltalike_freq_mismatch_annual(self, other):
+        rng = pd.period_range('2014', '2024', freq='A')
+        msg = ('Input has different freq(=.+)? '
+               'from PeriodIndex\\(freq=A-DEC\\)')
+        with tm.assert_raises_regex(period.IncompatibleFrequency, msg):
+            rng + other
+        with tm.assert_raises_regex(period.IncompatibleFrequency, msg):
+            rng += other
+
+    @pytest.mark.parametrize('other', [
+        pd.offsets.YearBegin(2),
+        pd.offsets.MonthBegin(1),
+        pd.offsets.Minute(),
+        np.timedelta64(365, 'D'),
+        timedelta(365)])
+    def test_pi_sub_isub_timedeltalike_freq_mismatch_annual(self, other):
+        rng = pd.period_range('2014', '2024', freq='A')
+        msg = ('Input has different freq(=.+)? '
+               'from PeriodIndex\\(freq=A-DEC\\)')
+        with tm.assert_raises_regex(period.IncompatibleFrequency, msg):
+            rng - other
+        with tm.assert_raises_regex(period.IncompatibleFrequency, msg):
+            rng -= other
+
+    def test_pi_add_iadd_timedeltalike_monthly(self):
+        rng = pd.period_range('2014-01', '2016-12', freq='M')
+        expected = pd.period_range('2014-06', '2017-05', freq='M')
+
+        result = rng + pd.offsets.MonthEnd(5)
+        tm.assert_index_equal(result, expected)
+
+        rng += pd.offsets.MonthEnd(5)
+        tm.assert_index_equal(rng, expected)
+
+    @pytest.mark.parametrize('other', [
+        pd.offsets.YearBegin(2),
+        pd.offsets.MonthBegin(1),
+        pd.offsets.Minute(),
+        np.timedelta64(365, 'D'),
+        timedelta(365),
+        Timedelta(days=365)])
+    def test_pi_add_iadd_timedeltalike_freq_mismatch_monthly(self, other):
+        rng = pd.period_range('2014-01', '2016-12', freq='M')
+        msg = 'Input has different freq(=.+)? from PeriodIndex\\(freq=M\\)'
+        with tm.assert_raises_regex(period.IncompatibleFrequency, msg):
+            rng + other
+        with tm.assert_raises_regex(period.IncompatibleFrequency, msg):
+            rng += other
+
+    @pytest.mark.parametrize('other', [
```

Comment:
so I'l like to see you not repeat the parameterization on every test. you can do this in a number of ways

- a fixture
- make a list of the parameters (then its easy to add on a particular test)